# Importing Relevant Libraries

In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score


# Importing the dataset and understanding it 

In [12]:
df_init = pd.read_csv("Advertising.csv")
df_init.head()

,Unnamed: 0,TV,Radio,Newspaper,Sales
0,1,230.1,37.8,69.2,22.1
1,2,44.5,39.3,45.1,10.4
2,3,17.2,45.9,69.3,9.3
3,4,151.5,41.3,58.5,18.5
4,5,180.8,10.8,58.4,12.9


## Checking for missing values and duplicates

In [14]:
df_init.isna().sum()

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

In [26]:
df_init.duplicated().sum()

0

Key Insights
- There are no missing values and neither are there duplicates.

In [30]:
df_init.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


In [44]:
df_init.columns

Index(['TV', 'Radio', 'Newspaper', 'Sales'], dtype='object')

In [54]:
df = df_init[["TV", "Radio", "Newspaper", "Sales"]].copy()

In [56]:
df.shape

(200, 4)

## Getting Descriptive Statistics 

In [63]:
df.describe(include="all")

,TV,Radio,Newspaper,Sales
count,200.000000,200.000000,200.000000,200.000000
mean,147.042500,23.264000,30.554000,14.022500
std,85.854236,14.846809,21.778621,5.217457
min,0.700000,0.000000,0.300000,1.600000
25%,74.375000,9.975000,12.750000,10.375000
50%,149.750000,22.900000,25.750000,12.900000
75%,218.825000,36.525000,45.100000,17.400000
max,296.400000,49.600000,114.000000,27.000000


Key Insights
- The counts are consistent with no missing values.
- For the TV:
      - 147.04 is spent via TV advertisement if you pick an advert at random.
- Radio:
      - 23.26 is spent via Radio advertisement if you pick an advert at random.
- Newspaper:
      - 14.02 is spent via Newspaper advertisement if you pick an advert at random.

- People spend more on Television advertisements on average as compared to Radio and Newspapers.